integrations/opsgenie/client.py

```diff
@@ -40,109 +41,130 @@ async def _get_single_resource(
+
         if cache := event.attributes.get(cache_key):
+            logger.debug(f"Returning on-call team {identifier} from cache")
             return cache
+
         api_version = await self.get_resource_api_version(ObjectKind.TEAM)
         url = f"{self.api_url}/{api_version}/teams/{identifier}"
         oncall_team = (await self._get_single_resource(url))["data"]
         event.attributes[cache_key] = oncall_team
+        logger.debug(f"Fetched and cached on-call team {identifier}")
         return oncall_team
 
     async def get_oncall_user(self, schedule_identifier: str) -> dict[str, Any]:
+        logger.debug(f"Fetching on-call user for schedule {schedule_identifier}")
+        cache_key = f"{ObjectKind.SCHEDULE}-USER-{schedule_identifier}"
+
+        if cache := event.attributes.get(cache_key):
+            logger.debug(f"Returning on-call user {schedule_identifier} from cache")
+            return cache
+
         api_version = await self.get_resource_api_version(ObjectKind.SCHEDULE)
         url = f"{self.api_url}/{api_version}/schedules/{schedule_identifier}/on-calls?flat=true"
-        return (await self._get_single_resource(url))["data"]
+        oncall_user = (await self._get_single_resource(url))["data"]
+        event.attributes[cache_key] = oncall_user
+        logger.debug(f"Fetched and cached on-call user {schedule_identifier}")
+        return oncall_user
 
     async def get_schedule_by_team(
         self, team_identifier: str
     ) -> Optional[dict[str, Any]]:
-        schedules = []
-        async for schedule_batch in self.get_paginated_resources(ObjectKind.SCHEDULE):
-            schedules.extend(schedule_batch)
-        return next(
-            (
-                schedule
-                for schedule in schedules
-                if schedule["ownerTeam"]["id"] == team_identifier
-            ),
-            {},
-        )
-
-    async def get_associated_alerts(
-        self, incident_identifier: str
-    ) -> list[dict[str, Any]]:
-        cache_key = f"{ObjectKind.INCIDENT}-{incident_identifier}"
+        if not team_identifier:
+            return {}
+        cache_key = f"{ObjectKind.SCHEDULE}-{ObjectKind.TEAM}-{team_identifier}"
         if cache := event.attributes.get(cache_key):
             return cache
+        async for schedule_batch in self.get_paginated_resources(ObjectKind.SCHEDULE):
+            for schedule in schedule_batch:
+                if schedule["ownerTeam"]["id"] == team_identifier:
+                    event.attributes[cache_key] = schedule
+                    return schedule
 
-        api_version = await self.get_resource_api_version(ObjectKind.INCIDENT)
-        url = f"{self.api_url}/{api_version}/incidents/{incident_identifier}/associated-alert-ids"
-        associated_alerts = (await self._get_single_resource(url))["data"]
-        event.attributes[cache_key] = associated_alerts
-        return associated_alerts
+        return {}
 
     async def get_impacted_services(
```

Comment:
do we still need the impacted_services? isn't it enough to just relate it using the `id` of the service?